### DISCUSSION TITLE
STRU file

### DISCUSSION THREAD

**Discussion by ferroteam** [deepmodeling/abacus-develop#1223] [General]
Hello, I'm a new user of ABACUS. I noticed there are a lot of keywords used in ABACUS that mimic Quantum Espresso and VASP keywords.  So, I was wondering if there is a tool that can generate STRU file from cif file or convert POSCAR (vasp input) to STRU? Manually copy/paste coordinates is quite tiring procedure.

Thanks!

**Comment by hongriTianqi** [comment]
Hi, thanks for your question. One way to change from POSCAR to STRU is to use the [ASE-ABACUS](https://gitlab.com/1041176461/ase-abacus) interface. An example of the converting script is provided below:

```
from ase.io import read, write
from pathlib import Path

cs_dir = './'
cs_vasp = Path(cs_dir, 'POSCAR')
cs_atoms = read(cs_vasp, format='vasp')
cs_stru = Path(cs_dir, 'STRU')
pp = {'Al':'Al.PD04.PBE.UPF'}
basis = {'Al':'Al_gga_10au_100Ry_3s3p2d.orb'}
write(cs_stru, cs_atoms, format='abacus', pp=pp, basis=basis)
```
